### PR TITLE
making standard form field

### LINKED ISSUE #193: Change Requests - Standard Change Request Form Fields
**Background:**
In order to allow users to submit new change requests, we need a submission form. This form needs to be able to handle all the different types of change requests. There are 3 core types of change requests: standard (scope) change requests, stage gates, and activation change requests. The standard (scope) change requests have a specific set of form fields which we can put together in this one component.

---

**Behavior:**
The new change request standard change request form fields component should allows users to input the what, why, scope impact, budget impact, and timeline impact. Additional specifications and behaviors are as follows. They may be complex to implement, so don't be afraid to do a little Googling, a little Youtube searching, and of course, ask for help from @kevinyu328 or @jamescd18.
- The what and scope impact fields should be long input text fields
  - they should not be the exact size shown in the sketch, just a reasonably larger rectangle to allow users to comfortably type a sentence or two
  - both input fields should be required
- The why input group should allow users to select why options
  - Users must select at least one option from the why input group
  - Each why option should have an adjacent small input field
- The budget impact and timeline impact input fields should be small text fields
  - Both input fields should be required and restricted to numerical input values (numbers)
  - The budget impact field should be an input group with a prepend for a dollar sign
  - The timeline impact field should be an input group with an append for the unit of time in weeks

---

**Steps:**
- [x] Create the basic component
  - [x] Create a basic react component folder structure in the `src/components` folder
  - [x] Create the test file, component file, and ccs module file in the component folder
  - [x] Add basic TS functional component boilerplate to the component `.tsx` file
  - [x] Add basic test suite `describe()` with at least one `it()` test case to the test file
  - [x] Add the component to the `app-authenticated` component (so you can see the component when starting the app in development)
- [x] Add the desired functionality
  - [x] Search through [react-bootstrap](http://react-bootstrap.github.io) to find the correct template component code to use
  - [x] Add the input fields to the component
  - [x] Add label text for each input field
  - [x] Pull in dummy example data instead of fetching data from the API
  - [x] Configure the logic specified in the behavior section
  - [ ] Add `useState()` local variables to store state variables that need to change
- [x] Apply styling
  - [x] Check that the label text is styled like label text
  - [x] Check that the layout of the input fields roughly matches the sketch
- [x] Create test cases
  - [x] Test that the label text is visible
  - [ ] Test that the input fields are there
  - [ ] Test that the behavior logic is working properly
- [x] Open a PR with a screenshot of the component to get review and feedback
  - [x] Ensure you attach a screenshot when opening a PR
  - [x] Ensure you follow the [instructions in the contributor guide](https://github.com/Northeastern-Electric-Racing/PM-Dashboard-v2/blob/main/docs/ContributorGuide.md#creating-a-pull-request)
  - [x] Iterate and make changes based on feedback
  - [ ] Once approved for merging, attach a screenshot of the finished component to this issue

---

**Sketch:**
The standard change request form fields section is highlighted with a red box around it. Notice that there is no visual styling separation between the standard change request form fields and the rest of the form fields except for the spacing and layout. Ideally [flexbox](https://getbootstrap.com/docs/4.0/utilities/flex/) would be used to ensure responsiveness.
- Ignore the "info icon"s
- Ignore the documentation link input group
- Ignore the submit button
- Ignore the fact that the input field next to the why "other project / work package" option is drawn as a dropdown
- Ignore the fact that there aren't input fields next to every why option
Screen Shot 2021-05-10 at 2 26 01 PM

### PR DESCRIPTION
**Pull request by VerySeriousHireableAccount**
closes #193 Standard Change Request Form Fields

### CODE REVIEW

**Comment by VerySeriousHireableAccount** [comment]
image

File: src/components/change-requests/new-change-request/new-change-request-page/standard-form-field/standard-form-field.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import { Row, Col, Form, InputGroup, FormControl } from 'react-bootstrap';
+import { exampleAllWorkPackages } from '../../../../../test-support/test-data/work-packages.stub';
+import { wbsPipe } from '../../../../../shared/pipes';
+import styles from './standard-form-field.module.css';
+
+const StandardFormFields: React.FC = () => {
+  return (
+    <div className={'row'}>
+      <Form className={'px-4'}>
+        <Form.Group controlId="exampleForm.ControlTextarea1">
```
Comment (jamescd18) [inline src/components/change-requests/new-change-request/new-change-request-page/standard-form-field/standard-form-field.tsx:15]: replace placeholder controlIds with meaningful ids for each form field. This one could probably be "newCR.what" and other fields would have "what" replaced with the field name

File: src/components/change-requests/new-change-request/new-change-request-page/standard-form-field/standard-form-field.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+import { Row, Col, Form, InputGroup, FormControl } from 'react-bootstrap';
+import { exampleAllWorkPackages } from '../../../../../test-support/test-data/work-packages.stub';
+import { wbsPipe } from '../../../../../shared/pipes';
+import styles from './standard-form-field.module.css';
+
+const StandardFormFields: React.FC = () => {
+  return (
+    <div className={'row'}>
+      <Form className={'px-4'}>
```
Comment (jamescd18) [inline src/components/change-requests/new-change-request/new-change-request-page/standard-form-field/standard-form-field.tsx:14]: form should be constructed higher up in the component tree

File: src/components/change-requests/new-change-request/new-change-request-page/standard-form-field/standard-form-field.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+ */
+
+import { Row, Col, Form, InputGroup, FormControl } from 'react-bootstrap';
+import { exampleAllWorkPackages } from '../../../../../test-support/test-data/work-packages.stub';
+import { wbsPipe } from '../../../../../shared/pipes';
+import styles from './standard-form-field.module.css';
+
+const StandardFormFields: React.FC = () => {
+  return (
+    <div className={'row'}>
```
Comment (jamescd18) [inline src/components/change-requests/new-change-request/new-change-request-page/standard-form-field/standard-form-field.tsx:13]: add some margin on the top and bottom

File: src/components/change-requests/new-change-request/new-change-request-page/standard-form-field/standard-form-field.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        <Form.Group className={'px-4'}>
+          {['Estimation Error', 'School Work', 'Manufacturing Issues', 'Rules Compliance', 'Other Project/Work Package', 'Other'].map((type) => (
+            <Row key={type} className="mb-3">
+              <Form.Check
+                type="checkbox"
+                id={type}
+                label={type}
+              />
+              {(type === "Other Project/Work Package") &&
+                <Form.Control as="select" custom>
```
Comment (jamescd18) [inline src/components/change-requests/new-change-request/new-change-request-page/standard-form-field/standard-form-field.tsx:57]: add unique Ids to each input box

File: src/components/change-requests/new-change-request/new-change-request-page/standard-form-field/standard-form-field.tsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+                  {exampleAllWorkPackages.map((p) => (
+                    <option>{wbsPipe(p.wbsNum)}</option>
+                  ))}
+                </Form.Control>}
+              {(type === "Other") && <Form.Control type="text" />}
+            </Row>
+          ))}
+        </Form.Group>
+        Documentation Link
+        <Form.Control type="text" />
```
Comment (jamescd18) [inline src/components/change-requests/new-change-request/new-change-request-page/standard-form-field/standard-form-field.tsx:67]: add unique Id

File: src/components/change-requests/new-change-request/new-change-request-page/standard-form-field/standard-form-field.tsx [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+      </Form>
+    </div>
+
```
Comment (jamescd18) [inline src/components/change-requests/new-change-request/new-change-request-page/standard-form-field/standard-form-field.tsx:70]: remove extra line

File: src/components/change-requests/new-change-request/new-change-request-page/standard-form-field/standard-form-field.tsx [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+          {['Estimation Error', 'School Work', 'Manufacturing Issues', 'Rules Compliance', 'Other Project/Work Package', 'Other'].map((type) => (
+            <Row key={type} className="mb-3">
+              <Form.Check
+                type="checkbox"
+                id={type}
+                label={type}
+              />
+              {(type === "Other Project/Work Package") &&
+                <Form.Control as="select" custom>
+                  {exampleAllWorkPackages.map((p) => (
```
Comment (jamescd18) [inline src/components/change-requests/new-change-request/new-change-request-page/standard-form-field/standard-form-field.tsx:58]: needs key field set

**Review by jamescd18 — CHANGES_REQUESTED** [review]
This is amazing! Really well done! Just a few suggestions for customizing the IDs, adjusting the padding, and fixing some lint errors.

File: src/components/change-requests/new-change-request/new-change-request-page/standard-form-fields/standard-form-fields.test.tsx [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+    expect(screen.getByText('Why')).toBeInTheDocument();
+  });
+
+  it('renders the Documentation Link form field', () => {
+    renderComponent();
+
+    expect(screen.getByText('Documentation Link')).toBeInTheDocument();
+  });
+});
```
Comment (jamescd18) [inline src/components/change-requests/new-change-request/new-change-request-page/standard-form-fields/standard-form-fields.test.tsx:52]:
```suggestion
});

```

**Review by jamescd18 — APPROVED** [review]

**Pull request merged**